CONTRIBUTING.md

```diff
@@ -514,6 +515,49 @@ this API we will not break anyone, because they weren't able to use it. The file
 `allowed-breaking-changes.txt` in the root of the repo is an exclusion file that
 can be used in these cases.
 
+### Feature Flags
+
+Sometimes we want to introduce new breaking behavior because we believe this is
+the correct default behavior for the CDK. The problem of course is that breaking
+changes are only allowed in major versions and those are rare.
+
+To address this need, we have a feature flags pattern/mechanism. It allows us to
+introduce new breaking behavior which is disabled by default (so existing
+projects will not be affected) but enabled automatically for new projects
+created through `cdk init`.
+
+The pattern is simple:
+
+1. Define a new const under
+   [cx-api/lib/features.ts](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/cx-api/lib/features.ts)
+   with the name of the context key that **enables** this new feature (for
+   example, `ENABLE_STACK_NAME_DUPLICATES`). The context key should be in the
+   form `module.Type:feature` (e.g. `@aws-cdk/core:enableStackNameDuplicates`).
+2. Use `node.tryGetContext(cxapi.ENABLE_XXX)` to check if this feature is enabled
```

Comment:
I'm not yet entirely intimate with the relationship between core and cx-api. With that caveat...

Shouldn't we just expose the flags via `core`. This would mean taking a dependency on `cxapi` (I understand this is already the case transitively). Construct authors typically don't have to care about `cxapi` and now they would have to.

The experience would be smoother if they could `core.FeatureFlag.ENABLE_XXX`.

Reply:
We need it in cx-api since the CLI uses it too and does not have a dependency on @aws-cdk/core.